### PR TITLE
Upgrade CAPI and CAPO to v1.7.2 / v0.10.3

### PR DESCRIPTION
**Pull request by yankcrime**
Scenarios tested:

- [x] Can I create a cluster?
- [x] Can I remove a workload pool and create a new one with a different name (test the hack code)?  (_NB_ https://github.com/unikorn-cloud/unikorn/commit/5656e3a64eb75d7362d37392bb49cc3dfd5d63f7 will need merging once this is in)
- [x] Can I delete the cluster?

### CODE REVIEW

File: charts/cluster-api-cluster-openstack/templates/cluster.yaml [hunk 1]
```diff
@@ -37,10 +37,9 @@ metadata:
     # Let CAPO do this in its chosen order.
     argocd.argoproj.io/sync-options: Delete=false
 spec:
-  cloudName: {{ .Values.openstack.cloud }}
   identityRef:
     name: {{ include "cloudconfig.name" . }}
-    kind: Secret
+    cloudName: {{ .Values.openstack.cloud }}
```
Comment (spjmurray) [inline charts/cluster-api-cluster-openstack/templates/cluster.yaml:42]: Okay, now this is an intriguing change... references in Kubernetes are supposed to be well known defined types e.g. https://github.com/kubernetes/api/blob/master/core/v1/types.go#L6578

File: charts/cluster-api-cluster-openstack/templates/control-plane.yaml [hunk 2]
```diff
@@ -73,23 +73,27 @@ spec:
...
-      cloudName: {{ .Values.openstack.cloud }}
       identityRef:
         name: {{ include "cloudconfig.name" . }}
-        kind: Secret
+        cloudName: {{ .Values.openstack.cloud }}
       {{- if .Values.controlPlane.machine.disk }}
       rootVolume:
         availabilityZone: {{ .Values.openstack.volumeFailureDomain }}
-        diskSize: {{ .Values.controlPlane.machine.disk.size }}
+        sizeGiB: {{ .Values.controlPlane.machine.disk.size }}
```
Comment (spjmurray) [inline charts/cluster-api-cluster-openstack/templates/control-plane.yaml:88]: Note: the units do appear to be the same

```go
                disk := map[string]interface{}{
                        "size": machine.DiskSize.Value() >> 30,
                }
```

File: charts/cluster-api-cluster-openstack/templates/control-plane.yaml [hunk 3]
```diff
@@ -73,23 +73,27 @@ spec:
...
       {{- end }}
       {{- with $cluster := .Values.cluster -}}
         {{- with $metadata := $cluster.serverMetadata -}}
           {{- printf "serverMetadata:" | nindent 6 }}
-          {{- toYaml $metadata | nindent 8 }}
+          {{ range $k, $v := $metadata -}}
```
Comment (spjmurray) [inline charts/cluster-api-cluster-openstack/templates/control-plane.yaml:93]: I _think_, I may be more comfortable with

```
          {{ range $k, $v := $metadata -}}
            {{- printf "- key: %s" $k | nindent 6 }}
            {{- printf "  value: %s" $v | nindent 6 }}
          {{ end -}}
```

as the nesting is less horrible.  Not that mustache isn't just horrible to start with.

Comment (yankcrime) [reply charts/cluster-api-cluster-openstack/templates/control-plane.yaml:93]: Done, this now looks like:

```
      serverMetadata:
      - key: cluster
        value: capitest
      - key: organization
        value: nscale
      - key: project
        value: dev
```

**Review by spjmurray — CHANGES_REQUESTED** [review]
Don't think you've bumped the `cloud-api-cluster-openstack` `Chart.yaml`, thus it won't get released :D I can chalk that off on ~~Monday~~ Tuesday, enjoy your holiday!

**Comment by yankcrime** [comment]
@spjmurray I have now.

**Review by spjmurray — APPROVED** [review]
Gan on then. let's see what happens locally...

**Pull request merged**